src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -1223,6 +1245,120 @@ export default class SourceFileLinter {
 		}
 	}
 
+	#analyzeNewAndApplySettings(node: ts.NewExpression | ts.CallExpression) {
+		node?.arguments?.filter((arg) => ts.isObjectLiteralExpression(arg))
+			.flatMap((arg) => arg.properties)
+			.forEach((prop) => {
+				if (ts.isPropertyAssignment(prop) &&
+					(ts.isCallExpression(node) ||
+						this.#isPropertyBindingType(node, prop.name.getText()))
+				) {
+					if (ts.isObjectLiteralExpression(prop.initializer)) {
+						this.#analyzeModelDataTypes(prop.initializer);
+					} else {
+						this.#analyzeModelDataTypesBinding(prop);
+					}
+				}
+			});
+	}
+
+	#analyzeModelDataTypes(node: ts.ObjectLiteralExpression) {
+		node?.properties.forEach((prop) => {
+			if (!ts.isPropertyAssignment(prop)) {
+				return;
+			}
+
+			// Get the type property
+			let typeField;
+			if (ts.isObjectLiteralExpression(prop.initializer)) {
+				typeField = prop.initializer.properties.find((prop: ts.ObjectLiteralElementLike) => {
+					return ts.isPropertyAssignment(prop) &&
+						ts.isIdentifier(prop.name) &&
+						prop.name.text === "type";
+				});
+			} else if (ts.isStringLiteral(prop.initializer) &&
+				ts.isIdentifier(prop.name) && prop.name.text === "type" &&
+				// Whether it's a direct property, named "type" of the Control
+				// or type in property binding
+				!ts.isNewExpression(prop.parent.parent)) {
+				typeField = prop;
+			}
+
+			if (typeField && ts.isPropertyAssignment(typeField)) {
+				this.#analyzeModelTypeField(typeField.initializer);
+			}
+		});
+	}
+
+	#analyzeModelDataTypesBinding(node: ts.PropertyAssignment) {
+		if (ts.isStringLiteral(node.initializer) &&
+			node.initializer.text.startsWith("{") && node.initializer.text.endsWith("}")) {
+			const imports = this.sourceFile.statements
+				.filter((stmnt): stmnt is ts.ImportDeclaration =>
+					stmnt.kind === ts.SyntaxKind.ImportDeclaration)
+				.map((importNode) => {
+					return {
+						moduleName: (importNode.moduleSpecifier as ts.StringLiteral).text,
+						variableName: importNode.importClause?.name?.text,
+					} as RequireDeclaration;
+				});
+
+			const {start: nodePos} = getPositionsForNode({
+				node: node,
+				sourceFile: this.sourceFile,
+				resourcePath: this.resourcePath,
+			});
+			const bindingLinter = new BindingLinter(this.resourcePath, this.context);
+			bindingLinter.lintPropertyBinding(node.initializer.text, imports, nodePos);
+		}
+	}
+
+	/**
+	 * Identify if a constructor argument property is bindable
+	 *
+	 * Get the class declaration by going up the chain and then find the argument property
+	 * from there. Directly finding the type of the property is not possible from here as it's
+	 * missing some context.
+	*/
+	#isPropertyBindingType(node: ts.NewExpression, propName: string) {
+		const controlAmbientModule =
+			this.getSymbolModuleDeclaration(this.checker.getTypeAtLocation(node).symbol);
+		const classArg = (controlAmbientModule?.body as ts.ModuleBlock)?.statements
```

Comment:
Isn't body defined as `ts.ModuleBody`? Please refrain from using `as` unless there is really no other solution. This poses the risk of bugs and errors in the code as the following code might rely on the wrong type.

Reply:
Done